packages/ripgrep-workspace/src/common/search-workspace-protocol.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonRpcServer } from '@theia/core/lib/common/messaging/proxy-factory';
+
+export const ISearchWorkSpaceServer = Symbol('ISearchWorkSpaceServer');
+
+export const searchPath = '/services/search';
+
+export interface ISearchWorkSpaceServer extends JsonRpcServer<ISearchWorkSpaceClient> {
+    query(regExp: string): Promise<string>;
```

Comment:
Should I not be able to specify the path under which the search should happen?

Reply:
This is still very WIP @lmcbout is just trying to make a frontend/backend service.
So this is just for testing that...